File: solutions/java/sum-of-multiples/1/src/main/java/SumOfMultiples.java
import java.util.HashSet;
import java.util.Set;

class SumOfMultiples {

    Set<Integer> factors; 
    int num;

    SumOfMultiples(int number, int[] set) {
        factors = new HashSet<Integer>();
        for(int f : set) {
            factors.add(f);
        }
        num =  number;
    }

    int getSum() {
        Set<Integer> s = new HashSet<Integer>();
        int l=0;
        int sum=0;
        for(int f : factors){
            for(int m=f; m<num && m!=0; m+=f){
                s.add(m);
                if(l !=s.size()){
                    l++;
                    sum +=m;
                }
            }
        }
        return sum;
    }

}

File: solutions/dart/sum-of-multiples/1/lib/sum_of_multiples.dart
class SumOfMultiples {
  int sum(List<int> factors, int limit){
    var sm = 0;
    var mults = <int>{};
    for(var f in factors){
      for(var m=f; m<limit && m!=0; m+=f){
        if(!mults.contains(m)){
          sm += m;
          mults.add(m);
        }
      }
    }
    return sm;
  }
}
